src/boot/efi/linux.c

```diff
@@ -80,7 +80,10 @@ EFI_STATUS load_trusted_image(
         if (err != EFI_SUCCESS)
                 return log_error_status_stall(err, u"Error getting loaded image device path: %r", err);
 
-        if (!secure_boot_enabled())
+        /* If we are told the image will be trusted by the firmware, we expect loading to succeed without the
+         * need for any secure boot workarounds. This means that loading *will* fail with a security
+         * violation if that is not the case! */
+        if (is_trusted || !secure_boot_enabled())
```

Comment:
to me this reads backwards... if it's already trusted we should skip the firmware authentication...